openwisp_users/admin.py

```diff
@@ -185,6 +185,15 @@ class Meta(BaseUserCreationForm.Meta):
     class Media:
         js = ('admin/js/jquery.init.js', 'openwisp-users/js/addform.js')
 
+    def clean(self):
+        data = super().clean()
+        email = data.get('email')
+        if email and User.objects.filter(email__iexact=email).exists():
```

Comment:
you're forgetting to exclude the current user from the query. This means that validating the current user will fail. Please write a test for this case (ensuring that validating an existing user passes).

Reply:
Yes, since the code was implemented in the UserCreation Form, I didn't really think it was required, but I'll include it in the new patch.